app/models/spree/vendor_ability.rb

```diff
@@ -2,7 +2,7 @@ class Spree::VendorAbility
   include CanCan::Ability
 
   def initialize(user)
-    @vendor_ids = user.vendors.pluck(:id)
+    @vendor_ids = user.vendors.where(state: 'active').pluck(:id)
```

Comment:
You can use existing scope https://github.com/spree-contrib/spree_multi_vendor/blob/master/app/models/spree/vendor.rb#L42